src/components/common/Balance.js

```diff
@@ -0,0 +1,34 @@
+
+import React from 'react'
+
+// denomination of one near in minimal non divisible units (attoNears)
+const NEAR_NOMINATION = 10**18
+
+const REG = /(?=(\B)(\d{3})+$)/g;
+
+const Balance = (props) => {
+   let amount = props.amount / NEAR_NOMINATION
```

Comment:
float numbers generally shouldn't be used with money amounts. We have 128-bits of precision in amounts, JS floats at most have 53 bits of precision.

So all code working with amounts need to be rewritten operating either on strings or long numbers (https://github.com/indutny/bn.js/ is preferred cause already used elsewhere).

Note that for this specific case it looks like operating directly on decimal string is the easiest solution.